x-pack/plugins/banners/server/ui_settings.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { UiSettingsServiceSetup } from 'src/core/server';
+import { BannerConfiguration } from '../common';
+import { isHexColor } from './utils';
+
+export const registerSettings = (
+  uiSettings: UiSettingsServiceSetup,
+  config: BannerConfiguration
+) => {
+  uiSettings.register({
+    'banners:placement': {
+      name: i18n.translate('xpack.banners.settings.placement.title', {
+        defaultMessage: 'Banner placement',
+      }),
+      category: ['banner'],
+      order: 1,
+      type: 'select',
+      value: config.placement,
```

Comment:
The trick here is that I'm using the `kibana.yml` configuration value as the default value when registering the uiSettings (thanks to the recently added synchronous `context.config.get` API)

The main upside is a better user experience, as the value displayed as the default one in the advanced settings app is the 'real' default value that would be effectively used. It's also easier to retrieve the computed config as we only need to query the uiSettings API.